ext/nnstreamer/tensor_filter/tensor_filter_nnfw.c

```diff
@@ -37,45 +39,131 @@
 void init_filter_nnfw (void) __attribute__ ((constructor));
 void fini_filter_nnfw (void) __attribute__ ((destructor));
 
+typedef struct
+{
+  nnfw_tensorinfo i_in;
+  nnfw_tensorinfo i_out;
+  nnfw_session *session;
+  gchar *model_path;
+} nnfw_pdata;
+
+static void nnfw_close (const GstTensorFilterProperties * prop,
+    void **private_data);
+
 /**
  * @brief The standard tensor_filter callback
  */
-static int nnfw_open (const GstTensorFilterProperties * prop,
-    void **private_data)
+static int
+nnfw_open (const GstTensorFilterProperties * prop, void **private_data)
 {
-  return 0;
+  NNFW_STATUS status;
+  int err = 0;
+  nnfw_pdata *pdata;
+
+  if (*private_data != NULL) {
+    pdata = *private_data;
+    if (strcmp (prop->model_file, pdata->model_path)) {
+      nnfw_close (prop, private_data);  /* "reopen" */
+    } else {
+      return 1;
```

Comment:
TODO we have to handle 'already loaded' case.
e.g, return 1 makes assertion in tf & tf-lite now.

Reply:
Yup. That is to be handled at tensor_filter_common.c and tensor_filter.c 
You may create a backlog issue and resolve it later :)